### DISCUSSION TITLE
No script called tgtg_get_tokens

### DISCUSSION THREAD

**Discussion by vambier** [Chouffy/home_assistant_tgtg#42] [Q&A] [answered]
I installed home_assistant_tgtg using hacs, in the directory 'custom_components/tgtg' there are only 3 files : 
```
__init__.py
manifest.json
sensor.py
```

It says to run the tgtg_get_tokens script, but there's no such script in this directory. When I look into this repository the script is there though and also a script called tgtg_get_favorites_item_id.py

Any idea what I could be doing wrong?

**Comment by swin66** [comment]
I must admit I struggled with getting it set up too.
I thought this step should be on a separate PC/mac (as per step 3), and then as I couldn;t get the tokens script to run I followed the manual steps as outlined in this https://github.com/ahivert/tgtg-python
I had to copy and edit the config yam manually and it seemed to work
BTW you don't have to specify your favourites and it will pull them all in.
I'm now struggling with getting notification working

**Comment by Chouffy** [comment, marked as answer]
This script is at the root of the project, here: https://github.com/Chouffy/home_assistant_tgtg/blob/main/tgtg_get_tokens.py (or the [downloadable version here](https://raw.githubusercontent.com/Chouffy/home_assistant_tgtg/main/tgtg_get_tokens.py))
The `custom_components/tgtg` folder only contains files for Home Assistant
And indeed the `tgtg_get_tokens` needs to run on a separate PC !
